app/javascript/blacklight/checkbox_submit.js

```diff
@@ -1,134 +1,119 @@
-
-        var checkbox = $('<input type="checkbox">')
-          .addClass( options.cssClass )
-          .attr('id', options.cssClass + '_' + uniqueId);
-        var label = $('<label>')
-          .addClass( options.cssClass )
-          .attr('for', options.cssClass + '_' + uniqueId)
-          .attr('title', form.attr('title') || '');
-        var span = $('<span>');
-
-        label.append(checkbox);
-        label.append(' ');
-        label.append(span);
-
-        var checkboxDiv = $('<div class="checkbox" />')
-          .addClass(options.cssClass)
-          .append(label);
-
-        function updateStateFor(state) {
-            checkbox.prop('checked', state);
-            label.toggleClass('checked', state);
-            if (state) {
-               //Set the Rails hidden field that fakes an HTTP verb
-               //properly for current state action.
-               form.find('input[name=_method]').val('delete');
-               span.html(form.attr('data-present'));
-            } else {
-               form.find('input[name=_method]').val('put');
-               span.html(form.attr('data-absent'));
-            }
-          }
-
-        form.append(checkboxDiv);
-        updateStateFor(checked);
-
-        checkbox.click(function() {
-            span.html(form.attr('data-inprogress'));
-            label.attr('disabled', 'disabled');
-            checkbox.attr('disabled', 'disabled');
-
-            $.ajax({
-                url: form.attr('action'),
-                dataType: 'json',
-                type: form.attr('method').toUpperCase(),
-                data: form.serialize(),
-                error: function() {
-                   label.removeAttr('disabled');
-                   checkbox.removeAttr('disabled');
-                   options.error.call();
-                },
-                success: function(data, status, xhr) {
-                  //if app isn't running at all, xhr annoyingly
-                  //reports success with status 0.
-                  if (xhr.status != 0) {
-                    checked = ! checked;
-                    updateStateFor(checked);
-                    label.removeAttr('disabled');
-                    checkbox.removeAttr('disabled');
-                    options.success.call(form, checked, xhr.responseJSON);
-                  } else {
-                    label.removeAttr('disabled');
-                    checkbox.removeAttr('disabled');
-                    options.error.call();
-                  }
-                }
-            });
-
-            return false;
-        }); //checkbox.click
-
-
-      }); //this.each
-      return this;
-    };
-
-  $.fn.blCheckboxSubmit.defaults =  {
-            //cssClass is added to elements added, plus used for id base
-            cssClass: 'blCheckboxSubmit',
-            error: function() {
-              alert("Error");
-            },
-            success: function() {} //callback
-  };
-})(jQuery);
+class CheckboxSubmit {
```

Comment:
I think this would remove IE11 support. Would then someone using Sprockets / Webpacker need to provide appropriate shims?

I'd be ok with doing something along these lines, but calling it more of a 8.0 release.

Reply:
What here is not supported by IE11?

Reply:
Ah, geeze, no class support at all in IE11?  I'll add `babel-plugin-transform-es2015-classes`